src/prisma/schedule.ts

```diff
@@ -37,3 +39,108 @@ async function initiateSchedule(scheduleyear: number) {
   });
   return schedule;
 }
+
+async function updateCurrentSchedule(
+  scheduleId: any,
+  courses: CourseSectionInput[]
+) {
+  // Simply creates a new schedule instead of updating the current schedule
+
+  try {
+    const currentSchedule = await prisma.schedule.findFirst({
+      where: {
+        id: parseInt(scheduleId),
+      },
+    });
+
+    if (currentSchedule) {
+      // Delete course sections linked to original schedule
+      await prisma.courseSection.deleteMany({
+        where: {
+          scheduleId: parseInt(scheduleId),
+        },
+      });
+
+      // Delete original schedule
+      await prisma.schedule.delete({
+        where: {
+          id: parseInt(scheduleId),
+        },
+      });
+    }
+  } catch (error) {
+    return {
+      success: false,
+      message: 'Error deleting original schedule',
+      errors: error,
+    };
+  }
+
+  const currentYear = new Date().getFullYear();
+  let newSchedule;
```

Comment:
We want to pull the `schedule` from either the newest one or from the ID instead of assuming an year here.

Reply:
Okay so pretty much back to the first implementation then right?